wandb/apis/reports/report.py

```diff
@@ -67,13 +67,26 @@ def from_url(cls, url, api=None):
 
     @staticmethod
     def _url_to_report_id(url):
+        path_msg = "Path must be `entity/project/reports/report_title--report_id`"
         try:
-            report, *_ = url.split("?")
-            # If the report title ends in trailing space
-            report = report.replace("---", "--")
-            *_, report_id = report.split("--")
+            report_path, *_ = url.split("?")
+            report_path = report_path.replace("---", "--")
+
+            if "--" not in report_path:
+                raise ValueError(path_msg)
+
+            *_, report_id = report_path.split("--")
+            if len(report_id) == 0:
+                raise ValueError("Invalid report id")
+
+            # The base64 = is stripped from the report url in core app,
+            # so we need to add it back in here
+            report_id = report_id.strip()
+            if report_id[-1] != "=":
+                report_id += "="
```

Comment:
you may want to add 2 `==`, `view:10` is encoded as `VmlldzoxMA==`

from reading this post: https://stackoverflow.com/questions/2941995/python-ignore-incorrect-padding-error-when-base64-decoding it seems to be safe to have extra `=`, so I'd recommend always adding 2 padding and decode/re-encode the report id

Reply:
Added and included different decoded view ids lengths in our tests :D